dafni_cli/commands/options.py

```diff
@@ -0,0 +1,231 @@
+                                and email address
+        license (str): URL to a license that applies to the dataset
+        rights (Optional[str]): Description of any usage rights, restrictions
+                                or citations required by users of the dataset
+        version_message (str): Version message
+    """
+
+    # By default click returns an empty tuple instead of None for options with
+    # multiple=True, override here for consistency
+    def default_none(ctx, _, value):
+        return None if len(value) == 0 else value
+
+    # Arguments that will be used to indicate something as required
+    # Will use the default required = False and have a default of None in the
+    # case all_optional is True
+    required_args = {"required": True}
+    if all_optional:
+        required_args = {"default": None}
+    required_args_tuple = {**required_args, "callback": default_none}
+
+    def decorator(function):
+        function = click.option(
+            "--title",
+            type=str,
+            **required_args,
+            help="Title of the dataset",
+        )(function)
+        function = click.option(
+            "--description",
+            type=str,
+            **required_args,
+            help="Description of the dataset",
+        )(function)
+        function = click.option(
+            "--identifier",
+            type=str,
+            callback=default_none,
+            multiple=True,
+            help="Permanent URL of external identifier for this dataset (e.g. DOI). (Can have multiple)",
+        )(function)
+        function = click.option(
+            "--subject",
+            type=click.Choice(DATASET_METADATA_SUBJECTS),
+            **required_args,
+            help="Subject, one of those found at https://inspire.ec.europa.eu/metadata-codelist/TopicCategory",
+        )(function)
+        function = click.option(
+            "--theme",
+            type=click.Choice(DATASET_METADATA_THEMES),
+            callback=default_none,
+            multiple=True,
+            help="Theme, one of those found at https://inspire.ec.europa.eu/Themes/Data-Specifications/2892. Can have multiple.",
+        )(function)
+        function = click.option(
+            "--language",
+            type=click.Choice(DATASET_METADATA_LANGUAGES),
+            **required_args,
+            help="Language",
+        )(function)
+        function = click.option(
+            "--keyword",
+            type=str,
+            **required_args_tuple,
+            multiple=True,
+            help="Keyword used in data searches (Can have multiple)",
+        )(function)
+        function = click.option(
+            "--standard",
+            type=(str, URLParamType(optional=True)),
+            default=None,
+            help="Name and URL of a standard to which this dataset conforms (e.g. www.iso.org/standard/39229.html). Either value may be empty using "
+            ".",
```

Comment:
I don't know if this is due to line length, but this "" doesn't show up in the help

Reply:
Ah yes, I clearly missed out the \\", so it autoformatted like that as joining two separate strings when I expected it to format it as `'Name and URL of a standard to which this dataset conforms (e.g. www.iso.org/standard/39229.html). Either value may be empty using "".'` instead.